networkx/algorithms/tests/test_swap.py

```diff
@@ -123,3 +123,18 @@ def test_degree_seq_c4():
     degrees = sorted(d for n, d in G.degree())
     G = nx.double_edge_swap(G, 1, 100)
     assert degrees == sorted(d for n, d in G.degree())
+
+
+def test_no_edges():
+    G = nx.DiGraph()
+    G.add_nodes_from([0, 1, 2])
+    pytest.raises(nx.NetworkXError, nx.directed_edge_swap, G)
+    G = nx.Graph()
+    G.add_nodes_from([0, 1, 2, 3])
+    pytest.raises(nx.NetworkXError, nx.double_edge_swap, G)
+
```

Comment:
Just another minor suggestion: using `pytest.raises` as a context manager with the `match=` check is a really nice feature that allows us to better test we're hitting the exact exception branch that we expect.

Reply:
great!